From c0709b23628fc442f4c3e82b3886d1c4b6b92b4e Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Thu, 14 Jun 2018 13:19:00 +0200
Subject: [PATCH 06/13] Add topic-miami devicetrees

Just a copy from the topic-miami branch contents, not actually added to Makefile

Changes:
Add USB PHY
MP: Rename "clkc" to "clk", fix pinmux for sdio0, disable FPGA load, USB PHY, USB otg, QSPI size
Larger u-boot and kernel partitions since "warrior"

Add topic_miamimp_defconfig (created using "bitbake -c savedefconfig virtual/kernel")
---
 arch/arm/boot/dts/topic-dyplo.dtsi            |  12 +
 arch/arm/boot/dts/topic-miami-dyplo-acp.dts   |  11 +
 arch/arm/boot/dts/topic-miami-dyplo.dts       |   3 +
 .../boot/dts/topic-miami-florida-gen-amp.dts  |  34 ++
 .../boot/dts/topic-miami-florida-gen-mio.dtsi |  18 +
 .../boot/dts/topic-miami-florida-gen-nand.dts |   3 +
 .../boot/dts/topic-miami-florida-gen-pt.dts   |   4 +
 arch/arm/boot/dts/topic-miami-florida-gen.dts |   2 +
 .../arm/boot/dts/topic-miami-florida-gen.dtsi | 266 ++++++++++
 .../dts/topic-miami-florida-mio-dyplo.dts     |   4 +
 .../topic-miami-florida-mio-nand-dyplo.dts    |   5 +
 arch/arm/boot/dts/topic-miami-florida-mio.dts |   2 +
 .../arm/boot/dts/topic-miami-florida-mio.dtsi | 142 +++++
 .../boot/dts/topic-miami-florida-pci-pt.dts   |   3 +
 arch/arm/boot/dts/topic-miami-florida-pci.dts |  12 +
 .../arm/boot/dts/topic-miami-florida-test.dts |  52 ++
 .../boot/dts/topic-miami-florida-test.dtsi    | 125 +++++
 .../dts/topic-miami-florida-xadc-test.dtsi    |  15 +
 arch/arm/boot/dts/topic-miami-florida.dtsi    |  27 +
 arch/arm/boot/dts/topic-miami-nand.dtsi       |  13 +
 arch/arm/boot/dts/topic-miami-vice.dts        |   2 +
 arch/arm/boot/dts/topic-miami.dtsi            | 334 ++++++++++++
 .../boot/dts/topic-miamilite-florida-test.dts |  32 ++
 arch/arm/boot/dts/topic-miamilite.dtsi        |  51 ++
 .../boot/dts/topic-miamiplus-florida-test.dts |  51 ++
 arch/arm/boot/dts/topic-miamiplus.dtsi        |  97 ++++
 .../zynqmp-topic-miamimp-florida-gen.dts      | 497 ++++++++++++++++++
 .../zynqmp-topic-miamimp-florida-test.dts     | 188 +++++++
 .../boot/dts/xilinx/zynqmp-topic-miamimp.dts  | 384 ++++++++++++++
 arch/arm64/configs/topic_miamimp_defconfig    | 399 ++++++++++++++
 30 files changed, 2788 insertions(+)
 create mode 100644 arch/arm/boot/dts/topic-dyplo.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-dyplo-acp.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-dyplo.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-gen-amp.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-gen-mio.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-gen-nand.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-gen-pt.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-gen.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-gen.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-mio-dyplo.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-mio-nand-dyplo.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-mio.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-mio.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-pci-pt.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-pci.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-test.dts
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-test.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-florida-xadc-test.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-florida.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-nand.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miami-vice.dts
 create mode 100644 arch/arm/boot/dts/topic-miami.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miamilite-florida-test.dts
 create mode 100644 arch/arm/boot/dts/topic-miamilite.dtsi
 create mode 100644 arch/arm/boot/dts/topic-miamiplus-florida-test.dts
 create mode 100644 arch/arm/boot/dts/topic-miamiplus.dtsi
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts
 create mode 100644 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
 create mode 100644 arch/arm64/configs/topic_miamimp_defconfig

diff --git a/arch/arm/boot/dts/topic-dyplo.dtsi b/arch/arm/boot/dts/topic-dyplo.dtsi
new file mode 100644
index 000000000000..c86e647109f6
--- /dev/null
+++ b/arch/arm/boot/dts/topic-dyplo.dtsi
@@ -0,0 +1,12 @@
+/* DTS snippet to add DYPLO to the kernel configuration. This assumes
+ * a default configuration of IRQ 89 and address 0x64400000 */
+&amba {
+	dyplo: dyplo@64400000 {
+		compatible = "topic,dyplo-1.00.a";
+		reg = <0x64400000 0x200000>;
+		interrupt-parent = <&intc>;
+		interrupts = <0 57 0x4>;
+		nvmem-cells = <&dyplolicense>;
+		nvmem-cell-names = "license";
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-dyplo-acp.dts b/arch/arm/boot/dts/topic-miami-dyplo-acp.dts
new file mode 100644
index 000000000000..a71bb019cba9
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-dyplo-acp.dts
@@ -0,0 +1,11 @@
+/dts-v1/;
+/include/ "topic-miami-florida-gen.dtsi"
+/include/ "topic-dyplo.dtsi"
+
+/*
+ * When Dyplo's DMA nodes are connected to the ACP port, everything is coherent
+ * so we tell the system that no cache maintenance is needed.
+ */
+&dyplo {
+	dma-coherent;
+};
diff --git a/arch/arm/boot/dts/topic-miami-dyplo.dts b/arch/arm/boot/dts/topic-miami-dyplo.dts
new file mode 100644
index 000000000000..f6e3de15f13f
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-dyplo.dts
@@ -0,0 +1,3 @@
+/dts-v1/;
+/include/ "topic-miami-florida-gen.dtsi"
+/include/ "topic-dyplo.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-gen-amp.dts b/arch/arm/boot/dts/topic-miami-florida-gen-amp.dts
new file mode 100644
index 000000000000..43d6db58047b
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-gen-amp.dts
@@ -0,0 +1,34 @@
+/* Devicetree for use with Asymetric Multiple Processing */
+/dts-v1/;
+/include/ "topic-miami-florida-gen.dtsi"
+
+/ {
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		/* Reserve the top 2 MB of the 0.5GB DDR RAM */
+		remoteproc_reserved: remoteproc@0x1fe00000 {
+			reg = <0x1fe00000 0x200000>;
+			no-map; /* For dma_declare_coherent_memory() */
+		};
+	};
+
+	/* Boot CPU1 with AMP software, using reserved memory for firmware */
+	zynq_remoteproc@1 {
+		compatible = "xlnx,zynq_remoteproc";
+		memory-region = <&remoteproc_reserved>;
+		interrupt-parent = <&intc>;
+		interrupts = <0 37 0>, <0 38 0>;
+		firmware = "amp1.elf";
+		/* vring0 is the ipino */
+		vring0 = <15>;
+		vring1 = <14>;
+	};
+};
+
+/* Remove the "alive" LED so it can be controlled by the AMP firmware */
+&led_alive {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida-gen-mio.dtsi b/arch/arm/boot/dts/topic-miami-florida-gen-mio.dtsi
new file mode 100644
index 000000000000..af36d0889f75
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-gen-mio.dtsi
@@ -0,0 +1,18 @@
+/ {
+		charger: battery-charger {
+			compatible = "gpio-charger";
+			charger-type = "mains";
+			gpios = <&gpioexf 6 1>; /* Pin 6 of IO expander, active low */
+		};
+};
+
+&i2cmux_2 {
+	/* Battery gauge */
+	fuelgauge: battery-gauge {
+		compatible = "ltc2943";
+		reg = <0x64>;
+		lltc,resistor-sense = <15>; /* 15 mOhm */
+		lltc,prescaler-exponent = <5>; /* 2^(2*5) = 1024 */
+		power-supplies = <&charger>;
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida-gen-nand.dts b/arch/arm/boot/dts/topic-miami-florida-gen-nand.dts
new file mode 100644
index 000000000000..a4f4a580f364
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-gen-nand.dts
@@ -0,0 +1,3 @@
+/dts-v1/;
+/include/ "topic-miami-florida-gen.dtsi"
+/include/ "topic-miami-nand.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-gen-pt.dts b/arch/arm/boot/dts/topic-miami-florida-gen-pt.dts
new file mode 100644
index 000000000000..846d46090cb3
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-gen-pt.dts
@@ -0,0 +1,4 @@
+/dts-v1/;
+/include/ "topic-miami-florida-gen.dtsi"
+/include/ "topic-miami-florida-xadc-test.dtsi"
+/include/ "topic-dyplo.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-gen.dts b/arch/arm/boot/dts/topic-miami-florida-gen.dts
new file mode 100644
index 000000000000..4fbfd8f196d8
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-gen.dts
@@ -0,0 +1,2 @@
+/dts-v1/;
+/include/ "topic-miami-florida-gen.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-gen.dtsi b/arch/arm/boot/dts/topic-miami-florida-gen.dtsi
new file mode 100644
index 000000000000..06dcd5964093
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-gen.dtsi
@@ -0,0 +1,266 @@
+/include/ "topic-miami-florida.dtsi"
+/include/ "topic-miami-florida-gen-mio.dtsi"
+
+/* On-board push buttons connected to EMIO GPIO 4 and 5 in reference design */
+&gpio_keys {
+	s2 {
+		label = "s2";
+		gpios = <&gpio0 58 1>; /* EMIO 4, active low */
+		linux,code = <60>; /* KEY_F2 */
+		gpio-key,wakeup;
+	};
+	s3 {
+		label = "s3";
+		gpios = <&gpio0 59 1>; /* EMIO 5, active low */
+		linux,code = <61>; /* KEY_F3 */
+		gpio-key,wakeup;
+	};
+};
+
+/* I2C bus layout */
+&i2cmux_0 {
+	/* HDMI OUT */
+	adv7511: adv7511@39 {
+		compatible = "adi,adv7511";
+		reg = <0x39>, <0x3f>;
+		reg-names = "primary", "edid";
+		adi,input-depth = <8>;
+		adi,input-colorspace = "rgb";
+		adi,input-clock = "1x";
+		adi,input-style = <1>;
+		adi,input-justification = "right";
+		adi,clock-delay = <0>;
+		#sound-dai-cells = <0>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			port@0 {
+				reg = <0>;
+				adv7511_in: endpoint {
+					remote-endpoint = <&axi_hdmi_out>;
+				};
+			};
+			port@1 {
+				reg = <1>;
+			};
+		};
+	};
+};
+
+&i2cmux_2 {
+	/* Touch */
+	touch: ad7879@2f {
+		compatible = "ad7879";
+		reg = <0x2F>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <64 0x2>; /* GPIO 64 (EMIO 10), falling edge */
+		touchscreen-swapped-x-y;
+		adi,resistance-plate-x = <620>;
+		touchscreen-max-pressure = <4096>;
+		adi,first-conversion-delay = /bits/ 8 <3>;
+		adi,acquisition-time = /bits/ 8 <1>;
+		adi,median-filter-size = /bits/ 8 <2>;
+		adi,averaging = /bits/ 8 <1>;
+		adi,conversion-interval = /bits/ 8 <255>;
+	};
+};
+
+&i2cmux {
+	i2c@1 {
+		/* HDMI O - Connects directly to HDMI connector */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+	};
+	i2c@3 {
+		/* HDMI IN */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <3>;
+		/* copied from zynq-zed-imageon.dts */
+		adv7611@4c {
+			compatible = "adi,adv7611";
+			reg = <0x4c>;
+			#address-cells = <0x1>;
+			#size-cells = <0x0>;
+			bus-width = <24>;
+			hsync-active = <0>; /* Parallel bus */
+			vsync-active = <0>;
+			interrupt-parent = <&gpio0>;
+			interrupts = <57 4>; /* EMIO GPIO 3 */
+			adi,int1-config = <1>; /* ADV76XX_INT1_CONFIG_ACTIVE_LOW */
+			reset-gpios = <&gpio0 55 0>; /* EMIO GPIO 1 */
+			port@0 {
+				reg = <0x0>;
+			};
+			port@1 {
+				reg = <0x1>;
+				adv7611: endpoint {
+					bus-width = <16>;
+					remote-endpoint = <&axi_hdmi_rx>;
+				};
+			};
+		};
+	};
+};
+
+&amba {
+	/* FPGA IP blocks */
+
+	axi_vdma_0: axivdma@43000000 {
+		compatible = "xlnx,axi-vdma-1.00.a";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		#dma-cells = <1>;
+		#dma-channels = <1>;
+		reg = <0x43000000 0x1000>;
+		dma-ranges = <0x00000000 0x00000000 0x40000000>;
+		xlnx,num-fstores = <3>;
+		xlnx,flush-fsync = <1>;
+		xlnx,addrwidth = <32>;
+		clocks = <&clkc 15>; /* FCLK0 */
+		clock-names = "s_axi_lite_aclk";
+		dma-channel@43000000 {
+			compatible = "xlnx,axi-vdma-mm2s-channel";
+			interrupts = <0 59 0x4>;
+			xlnx,datawidth = <64>;
+			xlnx,genlock-mode = <0x0>;
+			xlnx,include-dre = <0x0>;
+		};
+	};
+
+	axi_hdmi_0: axi_hdmi@70e00000 {
+		compatible = "adi,axi-hdmi-tx-1.00.a";
+		reg = <0x70e00000 0x10000>;
+		adi,is-rgb;
+		dmas = <&axi_vdma_0 0>;
+		dma-names = "video";
+		clocks = <&clockgen 3>;
+		port {
+			axi_hdmi_out: endpoint {
+				remote-endpoint = <&adv7511_in>;
+			};
+		};
+	};
+
+	axi_spdif_tx_0: axi-spdif-tx@0x75c00000 {
+		compatible = "adi,axi-spdif-tx-1.00.a";
+		reg = <0x75c00000 0x1000>;
+		dmas = <&dmac_s 0>;
+		dma-names = "tx";
+		clocks = <&clkc 15>, <&clockgen 1>, <&clockgen 1>;
+		clock-names = "axi", "ref", "spdif";
+
+		#sound-dai-cells = <0>;
+	};
+
+	axi_hdmi_rx_core: axi-hdmi-rx@43c10000 {
+		compatible = "adi,axi-hdmi-rx-1.00.a";
+		reg = <0x43c10000 0x10000>;
+		dmas = <&axi_dmac 0>;
+		dma-names = "rx";
+		gpios = <&gpioexf 3 0>; /* HPD, hotplug control (?) */
+
+		port {
+			#address-cells = <0>;
+			#size-cells = <0>;
+
+			axi_hdmi_rx: endpoint {
+				remote-endpoint = <&adv7611>;
+				bus-width = <24>;
+			};
+		};
+	};
+
+	axi_dmac: dmac@43c00000 {
+		compatible = "adi,axi-dmac-1.00.a";
+		reg = <0x43c00000 0x10000>;
+		#dma-cells = <1>;
+		interrupts = <0 58 0>;
+		clocks = <&clkc 16>;
+
+		adi,channels {
+			#size-cells = <0>;
+			#address-cells = <1>;
+			dma-channel@0 {
+				reg = <0>;
+				adi,length-width = <14>;
+				adi,source-bus-width = <64>;
+				adi,source-bus-type = <2>;
+				adi,destination-bus-width = <64>;
+				adi,destination-bus-type = <0>;
+				adi,2d;
+			};
+		};
+	};
+
+	axi_vdma_1: axivdma@43010000 {
+		compatible = "xlnx,axi-vdma-1.00.a";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		#dma-cells = <1>;
+		#dma-channels = <1>;
+		reg = <0x43010000 0x1000>;
+		dma-ranges = <0x00000000 0x00000000 0x40000000>;
+		xlnx,num-fstores = <3>;
+		xlnx,flush-fsync = <1>;
+		xlnx,addrwidth = <32>;
+		clocks = <&clkc 15>; /* FCLK0 */
+		clock-names = "s_axi_lite_aclk";
+		dma-channel@43000000 {
+			compatible = "xlnx,axi-vdma-mm2s-channel";
+			interrupts = <0 55 0x4>;
+			xlnx,datawidth = <64>;
+			xlnx,genlock-mode = <0x0>;
+			xlnx,include-dre = <0x0>;
+		};
+	};
+
+	axi_vdma_vga: axi_vdma_vga@7e000000 {
+		compatible = "topic,vdma-fb";
+		reg = <0x7e000000 0x10000>;
+		dmas = <&axi_vdma_1 0>;
+		dma-names = "video";
+		num-fstores = <3>;
+		width = <1024>;
+		height = <600>;
+		horizontal-sync = <136>;
+		horizontal-front-porch = <24>;
+		horizontal-back-porch = <160>;
+		horizontal-polarity = <0>;
+		vertical-sync = <8>;
+		vertical-front-porch = <4>;
+		vertical-back-porch = <23>;
+		vertical-polarity = <0>;
+	};
+};
+
+/ {
+	adv7511_hdmi_snd {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "HDMI monitor";
+		simple-audio-card,widgets =
+			"Speaker", "Speaker";
+		simple-audio-card,routing =
+			"Speaker", "TX";
+
+		simple-audio-card,dai-link@0 {
+			format = "spdif";
+			cpu {
+				sound-dai = <&axi_spdif_tx_0>;
+				frame-master;
+				bitclock-master;
+			};
+			codec {
+				sound-dai = <&adv7511>;
+			};
+		};
+	};
+};
+
+/* Make it so that the internal display is always fb0 */
+&aliases {
+	display0 = &axi_vdma_vga;
+	display1 = &axi_hdmi_0;
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida-mio-dyplo.dts b/arch/arm/boot/dts/topic-miami-florida-mio-dyplo.dts
new file mode 100644
index 000000000000..a94df2657f42
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-mio-dyplo.dts
@@ -0,0 +1,4 @@
+/dts-v1/;
+/include/ "topic-miami-florida-mio.dtsi"
+/include/ "topic-miami-florida-gen-mio.dtsi"
+/include/ "topic-dyplo.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-mio-nand-dyplo.dts b/arch/arm/boot/dts/topic-miami-florida-mio-nand-dyplo.dts
new file mode 100644
index 000000000000..d7183946320f
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-mio-nand-dyplo.dts
@@ -0,0 +1,5 @@
+/dts-v1/;
+/include/ "topic-miami-florida-mio.dtsi"
+/include/ "topic-miami-florida-gen-mio.dtsi"
+/include/ "topic-miami-nand.dtsi"
+/include/ "topic-dyplo.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-mio.dts b/arch/arm/boot/dts/topic-miami-florida-mio.dts
new file mode 100644
index 000000000000..5cf840f0ccfd
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-mio.dts
@@ -0,0 +1,2 @@
+/dts-v1/;
+/include/ "topic-miami-florida-mio.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-mio.dtsi b/arch/arm/boot/dts/topic-miami-florida-mio.dtsi
new file mode 100644
index 000000000000..e1e9c0f56a64
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-mio.dtsi
@@ -0,0 +1,142 @@
+/include/ "topic-miami.dtsi"
+
+/* Common components for MIO parts of Florida boards. These are
+ * accessible even without programming any logic. */
+
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+	};
+
+	/* USB PHY and reset IO pin */
+	usb_phy0: phy0 {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = <0>;
+		reset-gpios = <&gpioex 0 1>;
+	};
+
+	xtal_27Mhz: clock_xtal_27Mhz {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <27000000>;
+	};
+
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+	};
+};
+
+/* I2C bus layout */
+&i2c0 {
+	i2cmux: i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x70>;
+
+		i2cmux_0: i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+			/* Clock generator TI I2C chip, 27MHz . */
+			clockgen: cdce925pw@64 {
+				compatible = "ti,cdce925";
+				reg = <0x64>;
+				clocks = <&xtal_27Mhz>;
+				xtal-load-pf = <5>; /* Crystal load-capacitor value */
+				#clock-cells = <1>;
+				/* PLL options, optional. For example to get SSC 1%
+				 * centered, supply the following extra data:
+				 *PLL2 {
+				 *	spread-spectrum = <4>;
+				 *	spread-spectrum-center;
+				 *};
+				 */
+			};
+		};
+		i2cmux_2: i2c@2 {
+			/* GEN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <2>;
+			/* GPIO expander */
+			gpioexf: pca953x@20 {
+				compatible = "nxp,pca9534";
+				reg = <0x20>;
+				/* 0=USB_OTG_OC_N 1=USB_OTG_EN_N 2=n.c. ...*/
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				/* IRQ line connected to MIO 15 */
+				interrupt-parent = <&gpio0>;
+				interrupts = <15 0x8>; /* GPIO 15, IRQ_TYPE_LEVEL_LOW */
+			};
+			/* EEPROM (v3r1 board)*/
+			eepromc: eepromc@50 {
+				compatible = "at24,24c04";
+				reg = <0x50>;
+			};
+			/* Programmable clock */
+			si514: si514@55 {
+				reg = <0x55>;
+				#clock-cells = <0>;
+				compatible = "silabs,si514";
+			};
+		};
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	vbus-supply = <&reg_usb0_vbus>;
+	dr_mode = "otg";
+	usb-phy = <&usb_phy0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+};
+
+&pinctrl0 {
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO29", "MIO31", "MIO36";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO28", "MIO30", "MIO32", "MIO33", "MIO34",
+			       "MIO35", "MIO37", "MIO38", "MIO39";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida-pci-pt.dts b/arch/arm/boot/dts/topic-miami-florida-pci-pt.dts
new file mode 100644
index 000000000000..84f13a4c3a5a
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-pci-pt.dts
@@ -0,0 +1,3 @@
+/dts-v1/;
+/include/ "topic-miami-florida.dtsi"
+/include/ "topic-dyplo.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami-florida-pci.dts b/arch/arm/boot/dts/topic-miami-florida-pci.dts
new file mode 100644
index 000000000000..897b15517475
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-pci.dts
@@ -0,0 +1,12 @@
+/dts-v1/;
+/include/ "topic-miami-florida.dtsi"
+
+/* DTS snippet to add the PCI sender to the kernel configuration. */
+&amba {
+	pcisender@40000000 {
+		compatible = "topic,pcisender";
+		reg = <0x40000000 4096>;
+		interrupt-parent = <&intc>;
+		interrupts = <0 60 0x4>; /* Not used in driver (yet) */
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida-test.dts b/arch/arm/boot/dts/topic-miami-florida-test.dts
new file mode 100644
index 000000000000..ba887a306d78
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-test.dts
@@ -0,0 +1,52 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
+/include/ "topic-miami-florida-test.dtsi"
+
+&gpioexf {
+	/* IRQ line connected to MIO 15 */
+	interrupts = <15 0x8>; /* GPIO 15, IRQ_TYPE_LEVEL_LOW */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio15_default>;
+};
+
+
+/* The test board doesn't have PS JTAG, mux these pins as GPIO instead */
+&gpio0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio0_default>;
+};
+
+&pinctrl0 {
+	pinctrl_gpio0_default: gpio0-default {
+		mux {
+			function = "gpio0";
+			groups =
+				"gpio0_46_grp",	"gpio0_47_grp",	"gpio0_48_grp",
+				"gpio0_49_grp";
+		};
+
+		conf {
+			groups =
+				"gpio0_46_grp",	"gpio0_47_grp",	"gpio0_48_grp",
+				"gpio0_49_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+			bias-disable;
+		};
+	};
+
+	pinctrl_gpio15_default: gpio15-default {
+		mux {
+			function = "gpio0";
+			groups = "gpio0_15_grp";
+		};
+
+		conf {
+			groups = "gpio0_15_grp";
+			bias-pull-up;
+			bias-high-impedance;
+			io-standard = <1>;
+		};
+	};
+};
+
diff --git a/arch/arm/boot/dts/topic-miami-florida-test.dtsi b/arch/arm/boot/dts/topic-miami-florida-test.dtsi
new file mode 100644
index 000000000000..447005ca5070
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-test.dtsi
@@ -0,0 +1,125 @@
+/* Periferals on the florida-test board. Can be combined with miami and miamiplus */
+
+/* No ethernet on this board */
+&gem0 {
+	status = "disabled";
+};
+
+
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+	};
+};
+
+/* I2C bus layout */
+&i2c0 {
+	/* GPIO expander 3v3 */
+	gpioexf: pca953x@20 {
+		compatible = "nxp,pca9534";
+		reg = <0x20>;
+		/* 0=USB_OTG_OC_N 1=USB_OTG_EN_N 2=n.c. ...*/
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-controller;
+		interrupt-parent = <&gpio0>;
+	};
+
+	/* Programmable clock, to clock-capable pins on FPGA */
+	si514c: si514@carrier {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514carrier";
+	};
+
+	/* GPIO expanders 1v8 */
+	gpiotst1: pca953x@21 {
+		compatible = "nxp,pca9534";
+		reg = <0x21>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	gpiotst2: pca953x@22 {
+		compatible = "nxp,pca9534";
+		reg = <0x22>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	gpiotst3: pca953x@23 {
+		compatible = "nxp,pca9534";
+		reg = <0x23>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	vbus-supply = <&reg_usb0_vbus>;
+	dr_mode = "otg";
+	/* USB reset connected to output 0 of expander */
+	xlnx,phy-reset-gpio = <&gpioex 0 0>;
+	xlnx,phy-reset-gpio-tristate;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+};
+
+&pinctrl0 {
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO29", "MIO31", "MIO36";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO28", "MIO30", "MIO32", "MIO33", "MIO34",
+			       "MIO35", "MIO37", "MIO38", "MIO39";
+			bias-disable;
+		};
+	};
+};
+
+/* Fake dependency on clock to force it to run at 100Mhz instead of the
+ * factory preset. */
+&uart0 {
+    assigned-clocks = <&si514c>;
+    assigned-clock-rates = <100000000>;
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida-xadc-test.dtsi b/arch/arm/boot/dts/topic-miami-florida-xadc-test.dtsi
new file mode 100644
index 000000000000..aa78879ada72
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida-xadc-test.dtsi
@@ -0,0 +1,15 @@
+&spi1 {
+	status = "okay";
+	speed-hz = <100000000>;
+	/* DAC */
+	ad5065: ad5065@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		#clock-cells = <1>;
+		compatible = "ad5065";
+		/* SPI Setup */
+		reg = <0>;
+		spi-cpha;
+		spi-max-frequency = <50000000>;
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-florida.dtsi b/arch/arm/boot/dts/topic-miami-florida.dtsi
new file mode 100644
index 000000000000..1b79b5fc2c9a
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-florida.dtsi
@@ -0,0 +1,27 @@
+/include/ "topic-miami-florida-mio.dtsi"
+
+/* Common components for ALL florida boards */
+
+/* Ethernet PHY via EMIO. Logic contains a GMII to RGMII conversion, so
+ * the emac should see a GMII interface */
+&gem0 {
+	status = "okay";
+	phy-handle = <&gmiitorgmii>;
+	phy-mode = "gmii";
+
+	phy3: phy@3 {
+		/* PHY on the board */
+		device_type = "ethernet-phy";
+		reg = <0x3>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <65 0x8>; /* GPIO 65 (EMIO 11), IRQ_TYPE_LEVEL_LOW */
+		rxc-skew-ps = <1080>; /* Skew RX clock pad by 180 ps */
+		txc-skew-ps = <840>;  /* Skew TX clock pad by -60 ps */
+	};
+
+	gmiitorgmii: gmiitorgmii@8 {
+		compatible = "xlnx,gmii-to-rgmii-1.0";
+		reg = <0x8>;
+		phy-handle = <&phy3>;
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miami-nand.dtsi b/arch/arm/boot/dts/topic-miami-nand.dtsi
new file mode 100644
index 000000000000..a6ae7082ea8f
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-nand.dtsi
@@ -0,0 +1,13 @@
+/* Disables the QSPI controller and activates NAND instead. */
+
+&smcc {
+	status = "okay";
+};
+
+&nand0 {
+	status = "okay";
+};
+
+&qspi {
+	status = "disabled"; /* conflicts with NAND pins */
+};
diff --git a/arch/arm/boot/dts/topic-miami-vice.dts b/arch/arm/boot/dts/topic-miami-vice.dts
new file mode 100644
index 000000000000..e4cf68de9723
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami-vice.dts
@@ -0,0 +1,2 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
diff --git a/arch/arm/boot/dts/topic-miami.dtsi b/arch/arm/boot/dts/topic-miami.dtsi
new file mode 100644
index 000000000000..72e0fdfc4630
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miami.dtsi
@@ -0,0 +1,334 @@
+/include/ "zynq-7000.dtsi"
+
+/* Remove broken I2C driver and replace with bitbang GPIO */
+/delete-node/ &i2c0;
+/delete-node/ &i2c1;
+
+/ {
+	model = "Topic Zynq Miami";
+
+	interrupt-parent = <&intc>;
+
+	/* Arrange that UART0 is ttyPS0 */
+	aliases: aliases {
+		serial0 = &uart0;
+		serial1 = &uart1;
+		ethernet0 = &gem0;
+		mmc0 = &sdhci0;
+		mmc1 = &sdhci1;
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x000000000 0x20000000>;
+	};
+
+	chosen {
+		/* Boot using squashfs system, nice for QSPI NOR flash */
+		bootargs = "console=ttyPS0,115200 root=/dev/mtdblock5 ro rootfstype=squashfs rootwait quiet";
+		/* bootargs = "console=ttyPS0,115200 ubi.mtd=qspi-rootfs root=ubi0:qspi-rootfs rw rootfstype=ubifs rootwait"; */
+		linux,stdout-path = "/amba@0/uart@e0000000";
+	};
+
+	/* Represents the 3v3 supply on the Miami SOM. */
+	reg_3v3_miami: regulator_3v3_miami {
+		compatible = "regulator-fixed";
+		regulator-name = "3v3-miami";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_leds_default>;
+		led_alive: alive {
+			label = "alive:green";
+			gpios = <&gpio0 25 0>;
+			linux,default-trigger = "heartbeat";
+		};
+	};
+
+	i2c0: gpios-i2c@50 {
+		bus-id = <0>;
+		compatible = "i2c-gpio";
+		gpios = <&gpio0 51 0 /* sda */
+			 &gpio0 50 0 /* scl */
+			>;
+		i2c-gpio,delay-us = <3>;	/* 333 kHz (?) */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_i2c0_default>;
+	};
+
+	i2c1: gpios-i2c@52 {
+		bus-id = <1>;
+		compatible = "i2c-gpio";
+		gpios = <&gpio0 53 0 /* sda */
+			 &gpio0 52 0 /* scl */
+			>;
+		i2c-gpio,delay-us = <3>;	/* 333 kHz (?) */
+		#address-cells = <1>;
+		#size-cells = <0>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_i2c1_default>;
+	};
+};
+
+/* Disable the ARM timer. This clocksource does not compensate for frequency
+   scaling. The TTC timer does this, but has a lower rating so it would not
+   get selected as system clocksource.
+   Alternatives to disabling this clock are setting the source in sysfs or
+   adding clocksource=ttc_clocksource to the kernel bootargs */
+&global_timer {
+	status = "disabled";
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&sdhci0 {
+	disable-wp; /* We don't have a write-protect detection */
+	status = "okay";
+};
+
+
+&i2c1 {
+	/* AD converter for voltage monitoring */
+	monitorvolt: ad799x@29 {
+		compatible = "ad7999";
+		reg = <0x29>;
+		vcc-supply = <&reg_3v3_miami>;
+		vref-supply = <&reg_3v3_miami>;
+	};
+	/* GPIO expander */
+	gpioex: pca953x@41 {
+		compatible = "nxp,pca9536";
+		reg = <0x41>;
+		/* 0=USB_RESET 1=VTT_SHDWN_N 2=V_PRESENT 3=DEBUG_PRESENT */
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	/* Current monitoring chip */
+	monitorcurrent: ltc2990@4C {
+		compatible = "ltc2990";
+		reg = <0x4C>;
+	};
+	/* EEPROM */
+	eeprom: eeprom@50 {
+		compatible = "at24,24c04";
+		reg = <0x50>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		/* NVMEM entries */
+		gem0_mac: mac@1e8 {
+			reg = <0x1e8 6>;
+		};
+		gem1_mac: mac@1ee {
+			reg = <0x1ee 6>;
+		};
+		miamiserial: serial@1f4 {
+			reg = <0x1f4 4>;
+		};
+		dyplolicense: dyplolicense@1f8 {
+			reg = <0x1f8 8>;
+		};
+	};
+	/* IO Voltage controller */
+	ltc3562: ltc3562@65 {
+		compatible = "lltc,ltc3562";
+		reg = <0x65>; /* 0b1100101 */
+		regulators {
+			R400B_reg: R400B {
+				/* Supplies VCCO2 to bank 34 */
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <1800000>;
+				regulator-boot-on;
+				regulator-always-on;
+				lltc,operating-mode = <1>;
+			};
+			R600B_reg: R600B {
+				/* Supplies VCCO0 to Bank 13 */
+				regulator-min-microvolt = <3300000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+			/* R400A Not connected */
+			R600A_reg: R600A {
+				/* Supplies VCCO1 to bank 35 */
+				/* We actually want 1v8, but the wiring of
+				 * the regulator is weird so you can only get
+				 * 1768mV */
+				regulator-min-microvolt = <1768000>;
+				regulator-max-microvolt = <1768000>;
+				lltc,fb-voltage-divider = <316 100>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+		};
+	};
+};
+
+&nand0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_smc0_nand_default>;
+
+	/* Micron MT29F2G16ABBFAH4 timings */
+	arm,nand-cycle-t0 = <0x4>;
+	arm,nand-cycle-t1 = <0x4>;
+	arm,nand-cycle-t2 = <0x1>;
+	arm,nand-cycle-t3 = <0x2>;
+	arm,nand-cycle-t4 = <0x2>;
+	arm,nand-cycle-t5 = <0x2>;
+	arm,nand-cycle-t6 = <0x4>;
+
+	#address-cells = <0x1>;
+	#size-cells = <0x1>;
+	partition@nand-all {
+		label = "nand-all";
+		reg = <0x0 0x10000000>; /* 256 MB */
+	};
+	/* to mount as UBI:
+	   ubiattach /dev/ubi_ctrl -m 0
+	   ubimkvol /dev/ubi0 -N nand -m
+	   mkdir /media/nand
+	   mount -t ubifs /dev/ubi0_0 /media/nand
+	 */
+};
+
+&usb0 {
+	dr_mode = "otg";
+};
+
+&qspi {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_qspi0_default>;
+	is-dual = <0>;
+	num-cs = <1>;
+	primary_flash: ps7-qspi@0 {
+		compatible = "jedec,spi-nor";
+		/* Max frequency for quad read is 108MHz according to data sheet */
+		spi-max-frequency = <108000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0x0>;
+		partition@qspi-u-boot-spl {
+			label = "qspi-u-boot-spl";
+			reg = <0x00000 0x20000>;
+		};
+		partition@qspi-u-boot-img {
+			label = "qspi-u-boot-img";
+			reg = <0x20000 0x80000>;
+		};
+		partition@qspi-u-boot-env {
+			label = "qspi-u-boot-env";
+			reg = <0xA0000 0x20000>;
+		};
+		partition@qspi-device-tree {
+			label = "qspi-device-tree";
+			reg = <0xC0000 0x20000>;
+		};
+		partition@qspi-linux {
+			label = "qspi-linux";
+			reg = <0x100000 0x480000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x580000 0x1A80000>;
+		};
+		/* Everything */
+		partition@qspi-all {
+			label = "qspi-all";
+			reg = <0x0 0x2000000>;
+		};
+	};
+};
+
+&gem0 {
+	/* MAC address stored in NVMEM */
+	nvmem-cells = <&gem0_mac>;
+	nvmem-cell-names = "mac-address";
+};
+
+&gem1 {
+	/* MAC address stored in NVMEM */
+	nvmem-cells = <&gem1_mac>;
+	nvmem-cell-names = "mac-address";
+};
+
+&pinctrl0 {
+	pinctrl_qspi0_default: qspi0_default {
+		mux {
+			groups = "qspi0_0_grp";
+			function = "qspi0";
+		};
+		mux-fbclk {
+			groups = "qspi_fbclk_grp";
+			function = "qspi_fbclk";
+		};
+		conf {
+			groups = "qspi0_0_grp", "qspi_fbclk_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+			bias-disable;
+		};
+	};
+
+	pinctrl_smc0_nand_default: smc0_nand_default {
+		mux {
+			groups = "smc0_nand_grp";
+			function = "smc0_nand";
+		};
+	};
+
+	pinctrl_leds_default: leds-default {
+		mux {
+			groups = "gpio0_25_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_25_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+
+	pinctrl_i2c0_default: i2c0-default {
+		mux {
+			groups = "gpio0_50_grp", "gpio0_51_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_50_grp", "gpio0_51_grp";
+			bias-pull-up;
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+
+	pinctrl_i2c1_default: i2c1-default {
+		mux {
+			groups = "gpio0_52_grp", "gpio0_53_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_52_grp", "gpio0_53_grp";
+			bias-pull-up;
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miamilite-florida-test.dts b/arch/arm/boot/dts/topic-miamilite-florida-test.dts
new file mode 100644
index 000000000000..fe1e7339e2c0
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamilite-florida-test.dts
@@ -0,0 +1,32 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
+/include/ "topic-miamilite.dtsi"
+/include/ "topic-miami-florida-test.dtsi"
+
+&usb0 {
+	xlnx,phy-reset-gpio = <&gpio0 0 0>;
+};
+
+&gpioexf {
+	/* IRQ line connected to MIO 46 */
+	interrupts = <46 0x8>; /* GPIO 46, IRQ_TYPE_LEVEL_LOW */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio46_default>;
+};
+
+&pinctrl0 {
+	pinctrl_gpio46_default: gpio46-default {
+		mux {
+			function = "gpio0";
+			groups = "gpio0_46_grp";
+		};
+
+		conf {
+			groups = "gpio0_46_grp";
+			bias-pull-up;
+			bias-high-impedance;
+			io-standard = <1>;
+		};
+	};
+};
+
diff --git a/arch/arm/boot/dts/topic-miamilite.dtsi b/arch/arm/boot/dts/topic-miamilite.dtsi
new file mode 100644
index 000000000000..8b352dda67bc
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamilite.dtsi
@@ -0,0 +1,51 @@
+/* The "lite" version has less I/O pins and less periferals */
+
+/* Remove missing periferals */
+/delete-node/ &monitorcurrent;
+/delete-node/ &monitorvolt;
+/delete-node/ &ltc3562;
+/delete-node/ &gpioex;
+
+/* Alive LED is on MIO9 */
+&led_alive {
+	gpios = <&gpio0 9 1>; /* Active low */
+};
+
+&pinctrl_leds_default {
+	mux {
+		groups = "gpio0_9_grp";
+		function = "gpio0";
+	};
+	conf {
+		groups = "gpio0_9_grp";
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
+
+/* I2C bus 1 using pins 48+49 */
+&i2c1 {
+	gpios = <&gpio0 49 0 /* sda */
+		 &gpio0 48 0 /* scl */
+		>;
+   
+	/* 32Kb EEPROM */
+	eeprom: eeprom@50 {
+		compatible = "at24,24c32";
+		reg = <0x50>;
+	};
+};
+
+&pinctrl_i2c1_default {
+	mux {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		function = "gpio0";
+	};
+
+	conf {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		bias-pull-up;
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
diff --git a/arch/arm/boot/dts/topic-miamiplus-florida-test.dts b/arch/arm/boot/dts/topic-miamiplus-florida-test.dts
new file mode 100644
index 000000000000..43b917712308
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamiplus-florida-test.dts
@@ -0,0 +1,51 @@
+/dts-v1/;
+/include/ "topic-miami.dtsi"
+/include/ "topic-miamiplus.dtsi"
+/include/ "topic-miami-florida-test.dtsi"
+
+&gpioexf {
+	/* IRQ line connected to MIO 46 */
+	interrupts = <46 0x8>; /* GPIO 46, IRQ_TYPE_LEVEL_LOW */
+};
+
+/* The test board doesn't have ethernet, mux these pins as GPIO instead */
+&gpio0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio0_default>;
+};
+
+&pinctrl0 {
+	pinctrl_gpio0_default: gpio0-default {
+		mux {
+			function = "gpio0";
+			groups =
+				"gpio0_16_grp",	"gpio0_17_grp",	"gpio0_18_grp",
+				"gpio0_19_grp",	"gpio0_20_grp",	"gpio0_21_grp",
+				"gpio0_22_grp",	"gpio0_23_grp",	"gpio0_24_grp",
+				"gpio0_25_grp",	"gpio0_26_grp",	"gpio0_27_grp",
+				"gpio0_52_grp", "gpio0_53_grp";
+		};
+
+		conf {
+			groups =
+				"gpio0_16_grp",	"gpio0_17_grp",	"gpio0_18_grp",
+				"gpio0_19_grp",	"gpio0_20_grp",	"gpio0_21_grp",
+				"gpio0_22_grp",	"gpio0_23_grp",	"gpio0_24_grp",
+				"gpio0_25_grp",	"gpio0_26_grp",	"gpio0_27_grp",
+				"gpio0_52_grp", "gpio0_53_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-pull-none {
+			pins = "MIO52", "MIO53";
+			bias-disable;
+		};
+	};
+};
+
+/* Fake dependency on clock to force it to run at some frequency we'd like to see */
+&uart0 {
+	assigned-clock-rates = <103125000>;
+};
+
diff --git a/arch/arm/boot/dts/topic-miamiplus.dtsi b/arch/arm/boot/dts/topic-miamiplus.dtsi
new file mode 100644
index 000000000000..c63b30266fa3
--- /dev/null
+++ b/arch/arm/boot/dts/topic-miamiplus.dtsi
@@ -0,0 +1,97 @@
+/* the "plus" version has much more I/O pins, but lacks some periferals,
+ * and the NAND was replaced by a dual-QSPI NOR setup. */
+
+/* The miamiplus contains a speedgrade-2 device and runs at 800MHz */
+&cpu0 {
+	operating-points = <
+		/* kHz    uV */
+		800000  1000000
+		400000  1000000
+	>;
+};
+
+/* Remove missing periferals */
+/delete-node/ &monitorcurrent;
+/delete-node/ &ltc3562;
+
+/* Different voltage monitoring IC */
+&monitorvolt {
+	compatible = "ad7997";
+	reg = <0x21>;
+	/* TODO: gpios */
+};
+
+/* Dual QSPI configuration */
+&qspi {
+	is-dual = <1>;
+};
+
+/* Dual flash, also double the capacity so adjust rootfs and "all" partitions */
+&primary_flash {
+	partition@qspi-rootfs {
+		label = "qspi-rootfs";
+		reg = <0x500000 0x3b00000>;
+	};
+	partition@qspi-all {
+		label = "qspi-all";
+		reg = <0x0 0x4000000>;
+	};
+};
+
+/* The plus SOMs have a programmable clock for the high-speed tranceivers */
+&i2c1 {
+	si514s: si514@55 {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514miami";
+	};
+	max31760: max31760@57 {
+		reg = <0x57>; /* Address pins "111" hence address 1010111b */
+		compatible = "maxim,max31760";
+	};
+};
+
+/* Extra user board LED on GPIO expander */
+&leds {
+	led_user2: user2 {
+		label = "user2:green";
+		gpios = <&gpioex 3 0>;
+	};
+};
+/* Alive LED is on MIO7 instead of 25 */
+&led_alive {
+	gpios = <&gpio0 7 0>;
+};
+&pinctrl_leds_default {
+	mux {
+		groups = "gpio0_7_grp";
+		function = "gpio0";
+	};
+	conf {
+		groups = "gpio0_7_grp";
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
+
+/* I2C bus 1 using pins 48+49 */
+&i2c1 {
+	gpios = <&gpio0 49 0 /* sda */
+		 &gpio0 48 0 /* scl */
+		>;
+};
+
+&pinctrl_i2c1_default {
+	mux {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		function = "gpio0";
+	};
+
+	conf {
+		groups = "gpio0_48_grp", "gpio0_49_grp";
+		bias-pull-up;
+		slew-rate = <0>;
+		io-standard = <1>;
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
new file mode 100644
index 000000000000..e67323a15b2b
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
@@ -0,0 +1,497 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+#include "zynqmp-topic-miamimp.dts"
+
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+		regulator-always-on; /* for USB host mode, "vbus" no longer works? */
+	};
+
+	xtal_27Mhz: clock_xtal_27Mhz {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <27000000>;
+	};
+
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+		/* On-board push buttons in reference design */
+		s2 {
+			label = "s2";
+			gpios = <&gpio 95 1>; /* EMIO 17, active low */
+			linux,code = <60>; /* KEY_F2 */
+			gpio-key,wakeup;
+		};
+		s3 {
+			label = "s3";
+			gpios = <&gpio 96 1>; /* EMIO 18, active low */
+			linux,code = <61>; /* KEY_F3 */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+	};
+
+	charger: battery-charger {
+		compatible = "gpio-charger";
+		charger-type = "mains";
+		gpios = <&gpioexf 6 1>; /* Pin 6 of IO expander, active low */
+	};
+};
+
+&leds {
+	traffic_led_d4 { /* Red */
+		label = "d4:red";
+		gpios = <&gpio 82 0>; /* EMIO 4 */
+		linux,default-trigger = "mmc1";
+	};
+	traffic_led_d3 { /* Orange */
+		label = "d3:orange";
+		gpios = <&gpio 81 0>; /* EMIO 3 */
+		linux,default-trigger = "mmc0";
+	};
+	traffic_led_d2 { /* Green */
+		label = "d2:green";
+		gpios = <&gpio 80 0>; /* EMIO 2 */
+	};
+	traffic_led_d6 { /* Green */
+		label = "d6:green";
+		gpios = <&gpio 79 0>; /* EMIO 1 */
+	};
+	traffic_led_d5 { /* Green */
+		label = "d5:green";
+		gpios = <&gpio 78 0>; /* EMIO 0 */
+	};
+};
+
+/* SD card with level shifter */
+&sdhci1 {
+	status = "okay";
+	no-1-8-v;	/* for 1.0 silicon */
+	xlnx,mio_bank = <1>;
+	disable-wp; /* We don't have a write-protect detection */
+};
+
+&serdes {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+&sata {
+	status = "okay";
+	/* SATA OOB timing settings */
+	ceva,p0-cominit-params = /bits/ 8 <0x18 0x40 0x18 0x28>;
+	ceva,p0-comwake-params = /bits/ 8 <0x06 0x14 0x08 0x0E>;
+	ceva,p0-burst-params = /bits/ 8 <0x13 0x08 0x4A 0x06>;
+	ceva,p0-retry-params = /bits/ 16 <0x96A4 0x3FFC>;
+	ceva,p1-cominit-params = /bits/ 8 <0x18 0x40 0x18 0x28>;
+	ceva,p1-comwake-params = /bits/ 8 <0x06 0x14 0x08 0x0E>;
+	ceva,p1-burst-params = /bits/ 8 <0x13 0x08 0x4A 0x06>;
+	ceva,p1-retry-params = /bits/ 16 <0x96A4 0x3FFC>;
+	/* For now, only try one instance */
+	phy-names = "sata-phy", "sata-phy";
+	/* Use refclock 0 at 150MHz for both instances */
+	phys = <&lane0 PHY_TYPE_SATA 0 0 150000000>,
+	       <&lane1 PHY_TYPE_SATA 1 0 150000000>;
+	/* Using SI514 on miami through logic as REFCLK0. Must be programmed to 150MHz. */
+	assigned-clocks = <&si514s>;
+	assigned-clock-rates = <150000000>;
+	/* Apparently doesn't work this way? */
+	clocks = <&si514s>;
+};
+
+/* I2C bus layout */
+&i2c0 {
+	i2cmux: i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0x70>;
+
+		i2cmux_0: i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+			/* Clock generator TI I2C chip, 27MHz . */
+			clockgen: cdce925pw@64 {
+				compatible = "ti,cdce925";
+				reg = <0x64>;
+				clocks = <&xtal_27Mhz>;
+				xtal-load-pf = <5>; /* Crystal load-capacitor value */
+				#clock-cells = <1>;
+				/* PLL options, optional. For example to get SSC 1%
+				 * centered, supply the following extra data:
+				 *PLL2 {
+				 *	spread-spectrum = <4>;
+				 *	spread-spectrum-center;
+				 *};
+				 */
+			};
+			/* HDMI OUT */
+			adv7511: adv7511@39 {
+				compatible = "adi,adv7511";
+				reg = <0x39>, <0x3f>;
+				reg-names = "primary", "edid";
+				adi,input-depth = <8>;
+				adi,input-colorspace = "rgb";
+				adi,input-clock = "1x";
+				adi,input-style = <1>;
+				adi,input-justification = "right";
+				adi,clock-delay = <0>;
+				#sound-dai-cells = <0>;
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+					port@0 {
+						reg = <0>;
+						adv7511_in: endpoint {
+							remote-endpoint = <&axi_hdmi_out>;
+						};
+					};
+					port@1 {
+						reg = <1>;
+					};
+				};
+			};
+		};
+		i2cmux_2: i2c@2 {
+			/* GEN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <2>;
+			/* GPIO expander */
+			gpioexf: pca953x@20 {
+				compatible = "nxp,pca9534";
+				reg = <0x20>;
+				vcc-supply = <&reg_3v3_miami>;
+				gpio-line-names = "USB_OTG_OC_N", "USB_OTG_EN_N", "POWER_BTN", "HDMI_IN_HPD",
+					"", "POWER_KILL_N", "MAIN_POWER", "";
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				/* IRQ line connected to MIO ??? */
+				interrupt-parent = <&gpio>;
+				interrupts = <36 0x8>; /* GPIO 36, IRQ_TYPE_LEVEL_LOW */
+			};
+			/* Touch */
+			touch: ad7879@2f {
+				status = "disabled"; /* No display panel support, so touch is pointless */
+				compatible = "ad7879";
+				reg = <0x2F>;
+				interrupt-parent = <&gpio>;
+				interrupts = <94 0x2>; /* GPIO 78+16 (EMIO 16), falling edge */
+				touchscreen-swapped-x-y;
+				adi,resistance-plate-x = <620>;
+				touchscreen-max-pressure = <4096>;
+				adi,first-conversion-delay = /bits/ 8 <3>;
+				adi,acquisition-time = /bits/ 8 <1>;
+				adi,median-filter-size = /bits/ 8 <2>;
+				adi,averaging = /bits/ 8 <1>;
+				adi,conversion-interval = /bits/ 8 <255>;
+			};
+			/* EEPROM */
+			eepromc: eepromc@50 {
+				compatible = "at24,24c04";
+				reg = <0x50>;
+			};
+			/* Programmable clock */
+			si514c: si514@55 {
+				reg = <0x55>;
+				#clock-cells = <0>;
+				compatible = "silabs,si514";
+			};
+			/* Battery gauge */
+			fuelgauge: battery-gauge {
+				compatible = "ltc2943";
+				reg = <0x64>;
+				lltc,resistor-sense = <15>; /* 15 mOhm */
+				lltc,prescaler-exponent = <5>; /* 2^(2*5) = 1024 */
+				power-supplies = <&charger>;
+			};
+		};
+		i2c@3 {
+			/* HDMI IN */
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <3>;
+			/* copied from zynq-zed-imageon.dts */
+			adv7611@4c {
+				compatible = "adi,adv7611";
+				reg = <0x4c>;
+				#address-cells = <0x1>;
+				#size-cells = <0x0>;
+				bus-width = <24>;
+				hsync-active = <0>; /* Parallel bus */
+				vsync-active = <0>;
+				interrupt-parent = <&gpio>;
+				interrupts = <99 4>; /* EMIO 21 */
+				adi,int1-config = <1>; /* ADV76XX_INT1_CONFIG_ACTIVE_LOW */
+				reset-gpios = <&gpio 91 0>; /* EMIO 13 */
+				port@0 {
+					reg = <0x0>;
+				};
+				port@1 {
+					reg = <0x1>;
+					adv7611: endpoint {
+						bus-width = <16>;
+						remote-endpoint = <&axi_hdmi_rx>;
+					};
+				};
+			};
+		};
+	};
+};
+
+/* Ethernet: Micrel PHY via logic to RGMII, on MDIO#3 */
+&gem3 {
+	status = "okay";
+	phy-handle = <&gmiitorgmii>;
+	phy-mode = "gmii";
+	/* gem3 is the eth0 controller, hence we'll use the "0" mac for that */
+	nvmem-cells = <&gem0_mac>;
+	nvmem-cell-names = "mac-address";
+	phy3: phy@3 {
+		device_type = "ethernet-phy";
+		reg = <3>;
+		interrupt-parent = <&gpio>;
+		interrupts = <97 0x8>; /* GPIO 78+16+3, IRQ_TYPE_LEVEL_LOW */
+		rxc-skew-ps = <1080>; /* Skew RX clock pad by 180 ps */
+		txc-skew-ps = <840>;  /* Skew TX clock pad by -60 ps */
+	};
+	gmiitorgmii: gmiitorgmii@8 {
+		compatible = "xlnx,gmii-to-rgmii-1.0";
+		reg = <8>;
+		phy-handle = <&phy3>;
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	vbus-supply = <&reg_usb0_vbus>; /* This apparently no longer works */
+	dr_mode = "host"; /* "host", "peripheral" or "otg", apparently no effect */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+	usb-phy = <&usb_phy0>;
+	/* see: https://forums.xilinx.com/t5/Embedded-Linux/Zynqmp-USB2-0/td-p/790522/page/2 */
+	/delete-property/ clocks;
+	/delete-property/ clock-names;
+	clocks = <&clk 32>;
+	clock-names = "bus_clk";
+};
+&dwc3_0 {
+	status = "okay";
+	dr_mode = "host"; /* "host", "peripheral" or "otg" */
+	maximum-speed = "high-speed"; /* super-speed not available on carrier */
+	/* see: https://forums.xilinx.com/t5/Embedded-Linux/Zynqmp-USB2-0/td-p/790522/page/2 */
+	snps,dis_u2_susphy_quirk;
+	snps,dis_u3_susphy_quirk;
+	/delete-property/ phy-names;
+	/delete-property/ phys;
+};
+
+/* FPGA content (reference design) */
+&amba {
+	axi_vdma_0: axivdma@a0040000 {
+		compatible = "xlnx,axi-vdma-1.00.a";
+		reg = <0x0 0xa0040000 0x0 0x10000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		#dma-cells = <1>;
+		#dma-channels = <1>;
+		clock-names = "s_axi_lite_aclk", "m_axi_mm2s_aclk", "m_axi_mm2s_aclk";
+		clocks = <&clk 71>, <&clk 72>, <&clk 72>; /* fclk0 and fclk1 */
+		interrupt-parent = <&gic>;
+		interrupts = <0 89 4>;
+		dma-ranges = <0x00000000 0x00000000 0x80000000>;
+		xlnx,num-fstores = <3>;
+		xlnx,flush-fsync = <1>;
+		xlnx,addrwidth = <32>;
+		dma-channel@a0040000 {
+			compatible = "xlnx,axi-vdma-mm2s-channel";
+			interrupts = <0 89 4>; /* pl_ps_0[0] */
+			xlnx,datawidth = <64>;
+			xlnx,device-id = <0x0>;
+		};
+	};
+
+	axi_hdmi_0: axi_hdmi@a0030000 {
+		compatible = "adi,axi-hdmi-tx-1.00.a";
+		reg = <0x0 0xa0030000 0x0 0x10000>;
+		adi,is-rgb;
+		dmas = <&axi_vdma_0 0>;
+		dma-names = "video";
+		clocks = <&clockgen 3>;
+		port {
+			axi_hdmi_out: endpoint {
+				remote-endpoint = <&adv7511_in>;
+			};
+		};
+	};
+
+	axi_spdif_tx_0: axi-spdif-tx@a00f0000 {
+		status = "disabled"; /* DMA connection unknown */
+		compatible = "adi,axi-spdif-tx-1.00.a";
+		reg = <0x0 0xa00f0000 0x0 0x1000>;
+		/* TODO:  dmas = <&dmac_s 0>; */
+		dma-names = "tx";
+		clocks = <&clk 71>, <&clockgen 1>, <&clockgen 1>;
+		clock-names = "axi", "ref", "spdif";
+
+		#sound-dai-cells = <0>;
+	};
+
+
+	axi_hdmi_rx_core: axi-hdmi-rx@a0020000 {
+		compatible = "adi,axi-hdmi-rx-1.00.a";
+		reg = <0x0 0xa0020000 0x0 0x10000>;
+		dmas = <&axi_dmac 0>;
+		dma-names = "rx";
+		gpios = <&gpioexf 3 0>; /* HPD, hotplug control (?) */
+
+		port {
+			#address-cells = <0>;
+			#size-cells = <0>;
+
+			axi_hdmi_rx: endpoint {
+				remote-endpoint = <&adv7611>;
+				bus-width = <24>;
+			};
+		};
+	};
+
+	axi_dmac: dmac@a0000000 {
+		compatible = "adi,axi-dmac-1.00.a";
+		reg = <0x0 0xa0000000 0x0 0x10000>;
+		#dma-cells = <1>;
+		interrupt-parent = <&gic>;
+		interrupts = <0 90 4>; /* pl_ps_0[1] */
+		clocks = <&clk 71>; /* fclk0 */
+
+		adi,channels {
+			#size-cells = <0>;
+			#address-cells = <1>;
+			dma-channel@0 {
+				reg = <0>;
+				adi,length-width = <14>;
+				adi,source-bus-width = <64>;
+				adi,source-bus-type = <2>;
+				adi,destination-bus-width = <64>;
+				adi,destination-bus-type = <0>;
+				adi,2d;
+			};
+		};
+	};
+};
+
+/ {
+	adv7511_hdmi_snd {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "HDMI monitor";
+		simple-audio-card,widgets =
+			"Speaker", "Speaker";
+		simple-audio-card,routing =
+			"Speaker", "TX";
+
+		simple-audio-card,dai-link@0 {
+			format = "spdif";
+			cpu {
+				sound-dai = <&axi_spdif_tx_0>;
+				frame-master;
+				bitclock-master;
+			};
+			codec {
+				sound-dai = <&adv7511>;
+			};
+		};
+	};
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_sdhci1_default: sdhci1-default {
+		mux {
+			groups = "sdio1_4bit_0_1_grp";
+			function = "sdio1";
+		};
+
+		conf {
+			groups = "sdio1_4bit_0_1_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "sdio1_0_cd_grp";
+			function = "sdio1_cd";
+		};
+
+		conf-cd {
+			groups = "sdio1_0_cd_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-rx {
+			pins = "MIO52", "MIO53", "MIO55";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO54", "MIO56", "MIO57", "MIO58", "MIO59",
+			       "MIO60", "MIO61", "MIO62", "MIO63";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts
new file mode 100644
index 000000000000..d8bcf375ab71
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-test.dts
@@ -0,0 +1,188 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+#include "zynqmp-topic-miamimp.dts"
+
+/* Common components for MIO parts of Florida boards. These are
+ * accessible even without programming any logic. */
+/ {
+	/* Regulator for the 5V USB VBUS power line. This is connected
+	 * to the second output of the GPIO expander. */
+	reg_usb0_vbus: regulator_usb {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpioexf 1 0>;
+		enable-active-high;
+		regulator-always-on; /* USB host mode */
+	};
+
+	gpio_keys: gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		power {
+			label = "power";
+			gpios = <&gpioexf 2 1>; /* Input 2, active low */
+			linux,code = <116>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+		/* On-board push buttons in reference design */
+		s2 {
+			label = "s2";
+			gpios = <&gpio 95 1>; /* EMIO 17, active low */
+			linux,code = <60>; /* KEY_F2 */
+			gpio-key,wakeup;
+		};
+		s3 {
+			label = "s3";
+			gpios = <&gpio 96 1>; /* EMIO 18, active low */
+			linux,code = <61>; /* KEY_F3 */
+			gpio-key,wakeup;
+		};
+	};
+
+	gpio-poweroff {
+		compatible = "gpio-poweroff";
+		gpios = <&gpioexf 5 1>; /* Output 5, active low */
+		input;
+	};
+};
+
+/* SD card with level shifter */
+&sdhci1 {
+	status = "okay";
+	no-1-8-v;	/* for 1.0 silicon */
+	xlnx,mio_bank = <1>;
+	disable-wp; /* We don't have a write-protect detection */
+};
+
+&serdes {
+	status = "okay";
+};
+
+&uart1 {
+	status = "okay";
+};
+
+/* I2C bus layout */
+&i2c0 {
+	/* GPIO expander 3v3 */
+	gpioexf: pca953x@20 {
+		compatible = "nxp,pca9534";
+		reg = <0x20>;
+		/* 0=USB_OTG_OC_N 1=USB_OTG_EN_N 2=n.c. ...*/
+		gpio-controller;
+		#gpio-cells = <2>;
+		interrupt-controller;
+		interrupt-parent = <&gpio>;
+	};
+
+	/* Programmable clock, to clock-capable pins on FPGA */
+	si514c: si514@carrier {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514carrier";
+	};
+
+	/* GPIO expanders 1v8 */
+	gpiotst1: pca953x@21 {
+		compatible = "nxp,pca9534";
+		reg = <0x21>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	gpiotst2: pca953x@22 {
+		compatible = "nxp,pca9534";
+		reg = <0x22>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	gpiotst3: pca953x@23 {
+		compatible = "nxp,pca9534";
+		reg = <0x23>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+};
+
+/* USB connected through MIO and GPIO expander */
+&usb0 {
+	status = "okay";
+	vbus-supply = <&reg_usb0_vbus>; /* This apparently no longer works */
+	dr_mode = "otg"; /* "host", "peripheral" or "otg" */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+	usb-phy = <&usb_phy0>;
+};
+&dwc3_0 {
+	status = "okay";
+	dr_mode = "host"; /* "host", "peripheral" or "otg" */
+	maximum-speed = "high-speed"; /* super-speed not available on carrier */
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_sdhci1_default: sdhci1-default {
+		mux {
+			groups = "sdio1_4bit_0_1_grp";
+			function = "sdio1";
+		};
+
+		conf {
+			groups = "sdio1_4bit_0_1_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "sdio1_0_cd_grp";
+			function = "sdio1_cd";
+		};
+
+		conf-cd {
+			groups = "sdio1_0_cd_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-rx {
+			pins = "MIO52", "MIO53", "MIO55";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO54", "MIO56", "MIO57", "MIO58", "MIO59",
+			       "MIO60", "MIO61", "MIO62", "MIO63";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
new file mode 100644
index 000000000000..911a09d82751
--- /dev/null
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
@@ -0,0 +1,384 @@
+/*
+ * dts file for Topic Miami MPSoC
+ *
+ * (C) Copyright 2017, Topic Embedded Products BV
+ * Mike Looijmans <mike.looijmans@topic.nl>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or (at your option) any later version.
+ */
+
+/dts-v1/;
+
+#include "zynqmp.dtsi"
+#include "zynqmp-clk-ccf.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pinctrl/pinctrl-zynqmp.h>
+#include <dt-bindings/phy/phy.h>
+
+/ {
+	model = "Topic Miami MPSoC";
+	compatible = "topic,miamimp", "xlnx,zynqmp";
+
+	aliases {
+		ethernet0 = &gem3;
+		gpio0 = &gpio;
+		i2c0 = &i2c0;
+		i2c1 = &i2c1;
+		mmc0 = &sdhci0;
+		mmc1 = &sdhci1;
+		rtc0 = &rtc;
+		serial0 = &uart0;
+		serial1 = &uart1;
+		spi0 = &qspi;
+		usb0 = &usb0;
+	};
+
+	chosen {
+		bootargs = "earlycon root=/dev/mmcblk1p2 rw rootfstype=ext4 rootwait";
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+
+	/* Represents the 3v3 supply on the Miami SOM. */
+	reg_3v3_miami: regulator_3v3_miami {
+		compatible = "regulator-fixed";
+		regulator-name = "3v3-miami";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		heartbeat_led {
+			label = "heartbeat:green";
+			gpios = <&gpio 35 0>;
+			linux,default-trigger = "heartbeat";
+		};
+		user_led_2 {
+			label = "user2:green";
+			gpios = <&gpioex 3 0>;
+		};
+	};
+
+	/* USB PHY reset IO pin. Carrier board's USB PHY can use this. */
+	usb_phy0: phy0 {
+		compatible = "usb-nop-xceiv";
+		#phy-cells = <0>;
+		reset-gpios = <&gpioex 0 0>;
+	};
+
+	/* Full FPGA image */
+	fpga_full: fpga-full {
+		compatible = "fpga-region";
+		fpga-mgr = <&pcap>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		firmware-name = "fpga.bin";
+	};
+};
+
+&pcap {
+	/* It's broken and just hangs the board */
+	status = "disabled";
+};
+
+&gpio {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio_default>;
+};
+
+&gpu {
+	status = "okay";
+};
+
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c0_default>;
+	pinctrl-1 = <&pinctrl_i2c0_gpio>;
+	scl-gpios = <&gpio 26 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio 27 GPIO_ACTIVE_HIGH>;
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <400000>;
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_i2c1_default>;
+	pinctrl-1 = <&pinctrl_i2c1_gpio>;
+	scl-gpios = <&gpio 32 GPIO_ACTIVE_HIGH>;
+	sda-gpios = <&gpio 33 GPIO_ACTIVE_HIGH>;
+
+	/* GPIO expander */
+	gpioex: pca953x@41 {
+		compatible = "nxp,pca9536";
+		reg = <0x41>;
+		vcc-supply = <&reg_3v3_miami>;
+		gpio-line-names = "USB_RESET", "VTT_SHDWN_N", "V_PRESENT", "LED2";
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+	/* EEPROM */
+	eeprom: eeprom@50 {
+		compatible = "at24,24c04";
+		reg = <0x50>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		gem0_mac: mac@0x1e8 {
+			reg = <0x1e8 6>;
+		};
+		gem1_mac: mac@1ee {
+			reg = <0x1ee 6>;
+		};
+		miamiserial: serial@1f4 {
+			reg = <0x1f4 4>;
+		};
+		dyplolicense: dyplolicense@1f8 {
+			reg = <0x1f8 8>;
+		};
+	};
+	/* Programmable clock (for logic) */
+	si514s: si514@55 {
+		reg = <0x55>;
+		#clock-cells = <0>;
+		compatible = "silabs,si514";
+		clock-output-names = "si514miami";
+	};
+	/* AD converter for voltage monitoring */
+	monitorvolt: ad799x@21 {
+		compatible = "ad7997";
+		reg = <0x21>;
+		vcc-supply = <&reg_3v3_miami>;
+		vref-supply = <&reg_3v3_miami>;
+	};
+};
+
+&qspi {
+	status = "okay";
+	is-dual = <1>;
+	flash@0 {
+		compatible = "st,m25p80", "n25q256a";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0x0>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <4>;
+		spi-max-frequency = <166000000>;
+		is-dual = <1>;
+		partition@qspi-boot-bin {
+			label = "qspi-boot-bin";
+			reg = <0x00000 0x60000>;
+		};
+		partition@qspi-atf-spi-ub {
+			label = "qspi-atf-spi-ub";
+			reg = <0x60000 0x20000>;
+		};
+		partition@qspi-u-boot-bin {
+			label = "qspi-u-boot-bin";
+			reg = <0x80000 0xc0000>;
+		};
+		partition@qspi-u-boot-img {
+			label = "qspi-u-boot-img";
+			reg = <0x140000 0xc0000>;
+		};
+		partition@qspi-rootfs {
+			label = "qspi-rootfs";
+			reg = <0x200000 0x7e00000>;
+		};
+		/* Everything */
+		partition@qspi-all {
+			label = "qspi-all";
+			reg = <0x0 0x8000000>;
+		};
+	};
+};
+
+&rtc {
+	status = "okay";
+};
+
+/* eMMC device */
+&sdhci0 {
+	status = "okay";
+	non-removable;
+	disable-wp; /* We don't have a write-protect detection */
+	bus-width = <8>;
+	xlnx,mio_bank = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sdhci0_default>;
+};
+
+&serdes {
+	status = "okay";
+};
+
+&uart0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart0_default>;
+};
+
+&watchdog0 {
+	status = "okay";
+};
+
+&xilinx_ams {
+	status = "okay";
+};
+
+&ams_ps {
+	status = "okay";
+};
+
+&ams_pl {
+	status = "okay";
+};
+
+&pinctrl0 {
+	status = "okay";
+
+	pinctrl_i2c0_default: i2c0-default {
+		mux {
+			groups = "i2c0_6_grp";
+			function = "i2c0";
+		};
+
+		conf {
+			groups = "i2c0_6_grp";
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_i2c0_gpio: i2c0-gpio {
+		mux {
+			groups = "gpio0_26_grp", "gpio0_27_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_26_grp", "gpio0_27_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_i2c1_default: i2c1-default {
+		mux {
+			groups = "i2c1_8_grp";
+			function = "i2c1";
+		};
+
+		conf {
+			groups = "i2c1_8_grp";
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_i2c1_gpio: i2c1-gpio {
+		mux {
+			groups = "gpio0_32_grp", "gpio0_33_grp";
+			function = "gpio0";
+		};
+
+		conf {
+			groups = "gpio0_32_grp", "gpio0_33_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_uart0_default: uart0-default {
+		mux {
+			groups = "uart0_7_grp";
+			function = "uart0";
+		};
+
+		conf {
+			groups = "uart0_7_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-rx {
+			pins = "MIO30";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO31";
+			bias-disable;
+		};
+	};
+
+	pinctrl_sdhci0_default: sdhci0-default {
+		mux {
+			groups = "sdio0_0_grp";
+			function = "sdio0";
+		};
+
+		conf {
+			groups = "sdio0_0_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "sdio0_cd_0_grp";
+			function = "sdio0_cd";
+		};
+
+		conf-cd {
+			groups = "sdio0_cd_0_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+	};
+
+	pinctrl_gpio_default: gpio-default {
+		mux-sw {
+			function = "gpio0";
+			groups = "gpio0_35_grp";
+		};
+
+		conf-sw {
+			groups = "gpio0_35_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		mux-msp {
+			function = "gpio0";
+			groups = "gpio0_35_grp";
+		};
+
+		conf-msp {
+			groups = "gpio0_35_grp";
+			slew-rate = <SLEW_RATE_SLOW>;
+			io-standard = <IO_STANDARD_LVCMOS18>;
+		};
+
+		conf-pull-none {
+			pins = "MIO35";
+			bias-disable;
+		};
+	};
+};
diff --git a/arch/arm64/configs/topic_miamimp_defconfig b/arch/arm64/configs/topic_miamimp_defconfig
new file mode 100644
index 000000000000..66d293c764ad
--- /dev/null
+++ b/arch/arm64/configs/topic_miamimp_defconfig
@@ -0,0 +1,399 @@
+CONFIG_LOCALVERSION="-xilinx-v2018.1"
+CONFIG_SYSVIPC=y
+CONFIG_POSIX_MQUEUE=y
+CONFIG_AUDIT=y
+CONFIG_NO_HZ=y
+CONFIG_HIGH_RES_TIMERS=y
+CONFIG_BSD_PROCESS_ACCT=y
+CONFIG_TASKSTATS=y
+CONFIG_TASK_DELAY_ACCT=y
+CONFIG_TASK_XACCT=y
+CONFIG_TASK_IO_ACCOUNTING=y
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
+CONFIG_LOG_BUF_SHIFT=16
+CONFIG_CGROUPS=y
+CONFIG_BLK_DEV_INITRD=y
+# CONFIG_RD_BZIP2 is not set
+# CONFIG_RD_LZMA is not set
+CONFIG_EMBEDDED=y
+# CONFIG_COMPAT_BRK is not set
+CONFIG_SLAB=y
+CONFIG_PROFILING=y
+CONFIG_MODULES=y
+CONFIG_MODULE_UNLOAD=y
+CONFIG_ARCH_ZYNQMP=y
+CONFIG_NR_CPUS=8
+CONFIG_DEFAULT_MMAP_MIN_ADDR=32768
+CONFIG_TRANSPARENT_HUGEPAGE=y
+CONFIG_TRANSPARENT_HUGEPAGE_MADVISE=y
+CONFIG_CMA=y
+# CONFIG_DMI is not set
+CONFIG_COMPAT=y
+CONFIG_CPU_IDLE=y
+CONFIG_ARM_CPUIDLE=y
+CONFIG_CPU_FREQ=y
+CONFIG_CPU_FREQ_STAT=y
+CONFIG_CPU_FREQ_DEFAULT_GOV_ONDEMAND=y
+CONFIG_CPU_FREQ_GOV_USERSPACE=y
+CONFIG_CPUFREQ_DT=y
+CONFIG_NET=y
+CONFIG_PACKET=y
+CONFIG_UNIX=y
+CONFIG_XFRM_USER=y
+CONFIG_NET_KEY=y
+CONFIG_NET_KEY_MIGRATE=y
+CONFIG_INET=y
+CONFIG_IP_MULTICAST=y
+CONFIG_IP_PNP=y
+CONFIG_IP_PNP_DHCP=y
+CONFIG_IP_PNP_BOOTP=y
+CONFIG_IP_PNP_RARP=y
+CONFIG_SYN_COOKIES=y
+CONFIG_NETWORK_SECMARK=y
+CONFIG_NETFILTER=y
+CONFIG_NETFILTER_NETLINK_LOG=y
+CONFIG_NF_CONNTRACK=m
+CONFIG_NF_CT_NETLINK=m
+CONFIG_NETFILTER_XT_MARK=y
+CONFIG_NETFILTER_XT_TARGET_CHECKSUM=y
+CONFIG_NETFILTER_XT_TARGET_LOG=y
+CONFIG_NETFILTER_XT_TARGET_REDIRECT=m
+CONFIG_NETFILTER_XT_MATCH_CONNMARK=m
+CONFIG_NETFILTER_XT_MATCH_CONNTRACK=m
+CONFIG_NETFILTER_XT_MATCH_LIMIT=y
+CONFIG_NETFILTER_XT_MATCH_MAC=y
+CONFIG_NETFILTER_XT_MATCH_MULTIPORT=y
+CONFIG_NETFILTER_XT_MATCH_STATE=m
+CONFIG_NF_CONNTRACK_IPV4=m
+CONFIG_NF_NAT_IPV4=m
+CONFIG_IP_NF_IPTABLES=y
+CONFIG_IP_NF_FILTER=y
+CONFIG_IP_NF_TARGET_REJECT=y
+CONFIG_IP_NF_MANGLE=y
+CONFIG_NF_CONNTRACK_IPV6=m
+CONFIG_NF_NAT_IPV6=m
+CONFIG_IP6_NF_IPTABLES=y
+CONFIG_IP6_NF_FILTER=y
+CONFIG_IP6_NF_TARGET_REJECT=y
+CONFIG_IP6_NF_MANGLE=y
+CONFIG_BRIDGE_NF_EBTABLES=y
+CONFIG_BRIDGE_EBT_T_FILTER=y
+CONFIG_BRIDGE_EBT_T_NAT=y
+CONFIG_BRIDGE_EBT_MARK_T=y
+CONFIG_BRIDGE=y
+CONFIG_DNS_RESOLVER=y
+CONFIG_NET_PKTGEN=y
+CONFIG_CAN=m
+CONFIG_CAN_XILINXCAN=m
+CONFIG_BT=m
+CONFIG_BT_RFCOMM=y
+CONFIG_BT_RFCOMM_TTY=y
+CONFIG_BT_BNEP=y
+CONFIG_BT_BNEP_MC_FILTER=y
+CONFIG_BT_BNEP_PROTO_FILTER=y
+CONFIG_BT_HIDP=y
+CONFIG_BT_LEDS=y
+CONFIG_BT_HCIBTUSB=m
+CONFIG_BT_HCIBTSDIO=m
+CONFIG_BT_HCIUART=m
+CONFIG_BT_HCIUART_BCSP=y
+CONFIG_BT_HCIUART_ATH3K=y
+CONFIG_BT_HCIUART_LL=y
+CONFIG_BT_HCIUART_3WIRE=y
+CONFIG_BT_HCIUART_INTEL=y
+CONFIG_BT_HCIUART_QCA=y
+CONFIG_BT_HCIBCM203X=m
+CONFIG_BT_HCIBPA10X=m
+CONFIG_BT_HCIBFUSB=m
+CONFIG_BT_HCIVHCI=m
+CONFIG_BT_MRVL=m
+CONFIG_BT_MRVL_SDIO=m
+CONFIG_BT_ATH3K=m
+CONFIG_BT_WILINK=m
+CONFIG_CFG80211=m
+CONFIG_NL80211_TESTMODE=y
+CONFIG_CFG80211_CERTIFICATION_ONUS=y
+CONFIG_CFG80211_REG_CELLULAR_HINTS=y
+CONFIG_CFG80211_REG_RELAX_NO_IR=y
+CONFIG_CFG80211_WEXT=y
+CONFIG_MAC80211=m
+CONFIG_MAC80211_LEDS=y
+CONFIG_MAC80211_MESSAGE_TRACING=y
+CONFIG_MAC80211_DEBUG_MENU=y
+CONFIG_RFKILL=y
+CONFIG_RFKILL_INPUT=y
+CONFIG_RFKILL_GPIO=y
+CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
+CONFIG_DEVTMPFS=y
+CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_DMA_CMA=y
+CONFIG_CMA_SIZE_MBYTES=256
+CONFIG_ARM_CCI400_PMU=y
+CONFIG_CONNECTOR=y
+CONFIG_MTD=y
+CONFIG_MTD_CMDLINE_PARTS=y
+CONFIG_MTD_BLOCK=y
+CONFIG_MTD_CFI=y
+CONFIG_MTD_CFI_INTELEXT=y
+CONFIG_MTD_M25P80=y
+CONFIG_MTD_SPI_NOR=y
+# CONFIG_MTD_SPI_NOR_USE_4K_SECTORS is not set
+CONFIG_MTD_UBI=y
+CONFIG_MTD_UBI_FASTMAP=y
+CONFIG_MTD_UBI_BLOCK=y
+CONFIG_OF_OVERLAY=y
+CONFIG_OF_CONFIGFS=y
+CONFIG_BLK_DEV_LOOP=y
+CONFIG_BLK_DEV_LOOP_MIN_COUNT=4
+CONFIG_BLK_DEV_RAM=y
+CONFIG_BLK_DEV_RAM_COUNT=4
+CONFIG_BLK_DEV_RAM_SIZE=65536
+CONFIG_XILINX_JESD204B=m
+CONFIG_XILINX_JESD204B_PHY=m
+CONFIG_EEPROM_AT24=y
+CONFIG_EEPROM_AT25=m
+CONFIG_TI_ST=y
+CONFIG_BLK_DEV_SD=y
+CONFIG_BLK_DEV_SR=m
+CONFIG_ATA=y
+CONFIG_SATA_AHCI_PLATFORM=y
+CONFIG_AHCI_CEVA=m
+# CONFIG_ATA_SFF is not set
+CONFIG_NETDEVICES=y
+CONFIG_TUN=y
+# CONFIG_NET_VENDOR_ALACRITECH is not set
+# CONFIG_NET_VENDOR_AMAZON is not set
+# CONFIG_NET_VENDOR_AMD is not set
+# CONFIG_NET_VENDOR_AQUANTIA is not set
+# CONFIG_NET_VENDOR_ARC is not set
+CONFIG_MACB=m
+# CONFIG_NET_VENDOR_BROADCOM is not set
+# CONFIG_NET_VENDOR_EZCHIP is not set
+# CONFIG_NET_VENDOR_HISILICON is not set
+# CONFIG_NET_VENDOR_HUAWEI is not set
+# CONFIG_NET_VENDOR_INTEL is not set
+# CONFIG_NET_VENDOR_MARVELL is not set
+# CONFIG_NET_VENDOR_MELLANOX is not set
+# CONFIG_NET_VENDOR_MICREL is not set
+# CONFIG_NET_VENDOR_MICROCHIP is not set
+# CONFIG_NET_VENDOR_NATSEMI is not set
+# CONFIG_NET_VENDOR_NETRONOME is not set
+# CONFIG_NET_VENDOR_QUALCOMM is not set
+# CONFIG_NET_VENDOR_RENESAS is not set
+# CONFIG_NET_VENDOR_ROCKER is not set
+# CONFIG_NET_VENDOR_SAMSUNG is not set
+# CONFIG_NET_VENDOR_SEEQ is not set
+# CONFIG_NET_VENDOR_SOLARFLARE is not set
+# CONFIG_NET_VENDOR_SMSC is not set
+# CONFIG_NET_VENDOR_STMICRO is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
+# CONFIG_NET_VENDOR_XILINX is not set
+# CONFIG_NET_VENDOR_SYNOPSYS is not set
+CONFIG_MICREL_PHY=y
+CONFIG_XILINX_GMII2RGMII=m
+CONFIG_USB_USBNET=y
+# CONFIG_WLAN_VENDOR_ADMTEK is not set
+# CONFIG_WLAN_VENDOR_ATH is not set
+# CONFIG_WLAN_VENDOR_ATMEL is not set
+CONFIG_BRCMFMAC=m
+# CONFIG_WLAN_VENDOR_CISCO is not set
+# CONFIG_WLAN_VENDOR_INTEL is not set
+# CONFIG_WLAN_VENDOR_INTERSIL is not set
+# CONFIG_WLAN_VENDOR_MARVELL is not set
+# CONFIG_WLAN_VENDOR_MEDIATEK is not set
+# CONFIG_WLAN_VENDOR_RALINK is not set
+CONFIG_RTL8192CU=m
+# CONFIG_RTLWIFI_DEBUG is not set
+# CONFIG_WLAN_VENDOR_RSI is not set
+# CONFIG_WLAN_VENDOR_ST is not set
+# CONFIG_WLAN_VENDOR_TI is not set
+# CONFIG_WLAN_VENDOR_ZYDAS is not set
+# CONFIG_WLAN_VENDOR_QUANTENNA is not set
+CONFIG_INPUT_EVDEV=y
+CONFIG_KEYBOARD_GPIO=y
+# CONFIG_MOUSE_PS2 is not set
+CONFIG_VT_HW_CONSOLE_BINDING=y
+CONFIG_SERIAL_8250=m
+CONFIG_SERIAL_8250_RUNTIME_UARTS=2
+CONFIG_SERIAL_OF_PLATFORM=m
+CONFIG_SERIAL_UARTLITE=m
+CONFIG_SERIAL_XILINX_PS_UART=y
+CONFIG_SERIAL_XILINX_PS_UART_CONSOLE=y
+CONFIG_SERIAL_DEV_BUS=y
+CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
+# CONFIG_HW_RANDOM is not set
+CONFIG_I2C_CHARDEV=y
+CONFIG_I2C_MUX=y
+CONFIG_I2C_MUX_PCA9541=y
+CONFIG_I2C_MUX_PCA954x=y
+CONFIG_I2C_CADENCE=y
+CONFIG_I2C_XILINX=m
+CONFIG_SPI=y
+CONFIG_SPI_BITBANG=y
+CONFIG_SPI_CADENCE=m
+CONFIG_SPI_ZYNQMP_GQSPI=y
+CONFIG_PTP_1588_CLOCK=y
+CONFIG_GPIOLIB=y
+CONFIG_GPIO_SYSFS=y
+CONFIG_GPIO_XILINX=y
+CONFIG_GPIO_ZYNQ=y
+CONFIG_GPIO_PCA953X=y
+CONFIG_GPIO_PCA953X_IRQ=y
+CONFIG_GPIO_PCF857X=y
+CONFIG_POWER_RESET_GPIO=y
+CONFIG_POWER_RESET_LTC2952=y
+CONFIG_CHARGER_GPIO=y
+CONFIG_BATTERY_GAUGE_LTC2941=m
+CONFIG_SENSORS_IIO_HWMON=y
+CONFIG_PMBUS=y
+CONFIG_SENSORS_MAX20751=y
+CONFIG_WATCHDOG=y
+CONFIG_CADENCE_WATCHDOG=y
+CONFIG_MFD_TPS65086=y
+CONFIG_REGULATOR=y
+CONFIG_REGULATOR_FIXED_VOLTAGE=y
+CONFIG_REGULATOR_GPIO=y
+# CONFIG_RC_CORE is not set
+CONFIG_DRM=y
+CONFIG_DRM_ADI_AXI_HDMI=m
+CONFIG_DRM_PANEL_SIMPLE=y
+CONFIG_DRM_I2C_ADV7511=y
+CONFIG_DRM_I2C_ADV7511_AUDIO=y
+# CONFIG_DRM_I2C_ADV7533 is not set
+CONFIG_BACKLIGHT_LCD_SUPPORT=y
+CONFIG_BACKLIGHT_CLASS_DEVICE=y
+CONFIG_SOUND=y
+CONFIG_SND=y
+# CONFIG_SND_DRIVERS is not set
+# CONFIG_SND_SPI is not set
+CONFIG_SND_USB_AUDIO=m
+CONFIG_SND_SOC=y
+CONFIG_SND_SOC_XILINX_DP=m
+CONFIG_USB=y
+CONFIG_USB_ANNOUNCE_NEW_DEVICES=y
+# CONFIG_USB_DEFAULT_PERSIST is not set
+CONFIG_USB_OTG=y
+CONFIG_USB_OTG_FSM=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_STORAGE=y
+CONFIG_USB_UAS=y
+CONFIG_USB_DWC3=y
+CONFIG_USB_DWC3_ULPI=y
+CONFIG_NOP_USB_XCEIV=y
+CONFIG_USB_ULPI=y
+CONFIG_USB_GADGET=y
+CONFIG_USB_GADGET_XILINX=y
+CONFIG_USB_CONFIGFS=y
+CONFIG_USB_CONFIGFS_EEM=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
+CONFIG_USB_ETH=m
+CONFIG_USB_ETH_EEM=y
+CONFIG_USB_MASS_STORAGE=m
+CONFIG_USB_ULPI_BUS=y
+CONFIG_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ARASAN=y
+CONFIG_NEW_LEDS=y
+CONFIG_LEDS_CLASS=y
+CONFIG_LEDS_GPIO=y
+CONFIG_LEDS_TRIGGER_TIMER=y
+CONFIG_LEDS_TRIGGER_ONESHOT=y
+CONFIG_LEDS_TRIGGER_DISK=y
+CONFIG_LEDS_TRIGGER_MTD=y
+CONFIG_LEDS_TRIGGER_HEARTBEAT=y
+CONFIG_LEDS_TRIGGER_BACKLIGHT=y
+CONFIG_LEDS_TRIGGER_CPU=y
+CONFIG_LEDS_TRIGGER_GPIO=y
+CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
+CONFIG_LEDS_TRIGGER_TRANSIENT=y
+CONFIG_LEDS_TRIGGER_CAMERA=y
+CONFIG_LEDS_TRIGGER_PANIC=y
+CONFIG_EDAC=y
+CONFIG_EDAC_SYNOPSYS=y
+CONFIG_EDAC_ZYNQMP_OCM=y
+CONFIG_RTC_CLASS=y
+CONFIG_RTC_DRV_ZYNQMP=y
+CONFIG_DMADEVICES=y
+CONFIG_XILINX_DMA_ENGINES=y
+CONFIG_XILINX_DPDMA=m
+CONFIG_XILINX_FRMBUF=m
+CONFIG_XILINX_DMA=m
+CONFIG_XILINX_ZYNQMP_DMA=y
+CONFIG_DMATEST=m
+CONFIG_UIO=y
+CONFIG_UIO_PDRV_GENIRQ=m
+CONFIG_UIO_DMEM_GENIRQ=m
+CONFIG_UIO_XILINX_APM=m
+CONFIG_STAGING=y
+CONFIG_COMMON_CLK_XLNX_CLKWZRD=m
+CONFIG_XILINX_FCLK=y
+CONFIG_COMMON_CLK_SI514=y
+CONFIG_COMMON_CLK_SI544=y
+CONFIG_COMMON_CLK_SI570=y
+CONFIG_COMMON_CLK_CDCE925=y
+# CONFIG_COMMON_CLK_XGENE is not set
+CONFIG_COMMON_CLK_ZYNQMP=y
+CONFIG_ARM_SMMU=y
+CONFIG_REMOTEPROC=y
+CONFIG_ZYNQMP_R5_REMOTEPROC=m
+CONFIG_XILINX_VCU=m
+CONFIG_IIO=y
+CONFIG_AD799X=y
+CONFIG_XILINX_XADC=y
+CONFIG_XILINX_AMS=y
+CONFIG_RESET_CONTROLLER=y
+CONFIG_ZYNQMP_RESET_CONTROLLER=y
+CONFIG_PHY_XILINX_ZYNQMP=y
+CONFIG_RAS=y
+CONFIG_NVMEM_ZYNQMP=y
+CONFIG_FPGA=y
+CONFIG_ZYNQMP_FIRMWARE_DEBUG=y
+CONFIG_EXT4_FS=y
+CONFIG_EXT4_FS_POSIX_ACL=y
+CONFIG_EXT4_FS_SECURITY=y
+CONFIG_AUTOFS4_FS=y
+CONFIG_ISO9660_FS=m
+CONFIG_JOLIET=y
+CONFIG_UDF_FS=m
+CONFIG_MSDOS_FS=y
+CONFIG_VFAT_FS=y
+CONFIG_TMPFS=y
+CONFIG_TMPFS_POSIX_ACL=y
+CONFIG_HUGETLBFS=y
+CONFIG_JFFS2_FS=y
+CONFIG_JFFS2_SUMMARY=y
+CONFIG_JFFS2_FS_XATTR=y
+CONFIG_JFFS2_COMPRESSION_OPTIONS=y
+CONFIG_JFFS2_LZO=y
+CONFIG_JFFS2_RUBIN=y
+CONFIG_UBIFS_FS=m
+CONFIG_UBIFS_FS_ADVANCED_COMPR=y
+CONFIG_UBIFS_FS_ENCRYPTION=y
+CONFIG_SQUASHFS=y
+CONFIG_SQUASHFS_FILE_DIRECT=y
+CONFIG_SQUASHFS_DECOMP_MULTI=y
+CONFIG_SQUASHFS_LZO=y
+CONFIG_SQUASHFS_XZ=y
+# CONFIG_NETWORK_FILESYSTEMS is not set
+CONFIG_NLS_CODEPAGE_437=y
+CONFIG_NLS_ISO8859_1=y
+CONFIG_DEBUG_INFO=y
+CONFIG_DEBUG_FS=y
+CONFIG_MAGIC_SYSRQ=y
+# CONFIG_SCHED_DEBUG is not set
+# CONFIG_FTRACE is not set
+CONFIG_CRYPTO_ECDH=y
+# CONFIG_CRYPTO_MANAGER_DISABLE_TESTS is not set
+CONFIG_CRYPTO_CCM=y
+CONFIG_CRYPTO_GCM=y
+CONFIG_CRYPTO_CBC=y
+CONFIG_CRYPTO_ECB=y
+CONFIG_CRYPTO_CMAC=y
+CONFIG_CRYPTO_CRCT10DIF=y
+CONFIG_CRYPTO_MD5=y
+CONFIG_CRYPTO_ARC4=y
+# CONFIG_CRYPTO_HW is not set
+CONFIG_CRC7=y
-- 
2.17.1

